scipy/cluster/hierarchy.py

```diff
@@ -3196,6 +3253,12 @@ def llf(id):
     above_threshold_color : str, optional
         This matplotlib color string sets the color of the links above the
         color_threshold. The default is ``'C0'``.
+    leaves_order : iterable, optional
+        Plots the leaves in the order specified by a vector of
+        original observation indices. If the vector contains duplicates
+        or results in a crossing, an exception will be thrown. Passing
```

Comment:
How does a user know that a crossing will not happen, and if it happens what can they do about it?

Reply:
I've chosen an approach in which user must guarantee that the ordering is correct. As I've mentioned above, this function basically reorders the cut tree. If I understand everything correctly, in case provided ordering isn't correct user will get an exception from here:https://github.com/scipy/scipy/blob/8a6ac2cfb353dcd5be3edca9645ecf4bd32a9c23/scipy/cluster/hierarchy.py#L3280

Reply:
So I'm guessing that it's more common than not to get a crossing when a user puts in some random ordering, the envisioned workflow is:

- Use without `leaves_order`
- Review the plot, decide which items to swap
- Map node labels to indices to be used as leave ordering
- Add the right `leaves_order`

Without trying it, I'm still hesitant on how practical this is going to be.

Reply:
Hmm.. by 'envisioned workflow' You mean **user's** workflow, right?

Reply:
For me, this will actually be quite useful. In my use case, I'm looking at how behavior changes over time and combining time intervals based on the distance between each interval. Also, I want to keep the intervals in their inherent, temporal ordering.

Reply:
> by 'envisioned workflow' You mean user's workflow, right?

yes indeed

thanks @ihowell, that is helpful feedback.

Reply:
@rgommers @ihowell in the recent commit I've added full parameter correctness checking (e.g. if it's indeed a permutation, and if it doesn't result in a crossing). I'm now wondering whether it's possible (and reasonable) with my approach to support `truncate_mode` option. 
  I'm tempted to ask, whether it would be possible to say, that this new option is incompatible with `truncate_mode` parameter.
 The reason is: suppose one uses `truncate_mode` and gets some new leaves, which are in reality inner nodes. As far as I know, now the only way to label these nodes (which are now leaves) is to provide `leaf_label_func`. I don't see how in this case user should say that he wants to reorder these new leaves. Would be grateful if You express Your opinions.